docs/ADRs/0010-container-registry.md

```diff
@@ -0,0 +1,67 @@
+# ADR-0010 Container Registry
+
+## Context
+
+(Docker) images allow us to package and run applications in a standardized and portable way.
+To be able to use and share our own (custom) created images, they need to be stored in a remote registry
+that is accessible by others.
+
+There are many container registries. During research the following registries have been noted:
+
+[Docker Hub](https://hub.docker.com/),
+[GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry),
+[Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/),
+[Azure Container Registry (ACR)](https://azure.microsoft.com/en-us/products/container-registry),
+[Google Artifact Registry (GAR)](https://cloud.google.com/artifact-registry),
+[Red Hat Quay](https://www.quay.io/),
+[GitLab Container Registry](https://docs.gitlab.com/ee/user/packages/container_registry/),
+[Harbor](https://goharbor.io/),
+[Sonatype Nexus Repository Manager](https://www.sonatype.com/products/sonatype-nexus-repository),
+[JFrog Artifactory](https://jfrog.com/help/r/jfrog-artifactory-documentation).
+
+## Assumptions
+
+* We do not want to host our own registry.
+* The images we create can be kept private or publicly shared.
+* For development and testing, images should be kept private to prevent accidental use of unfinished products.
+* Images we provide are safe and secure. This means a container registry should have the option to (continuously) verify
+the security status of an image.
+* Tags are immutable, to prevent image tags from being overwritten.
```

Comment:
i think this is a way of working. We might want to overwrite a tag

Reply:
True. I have seen strange behaviour and expectations though with overwriting tags, which is why I mentioned this explicitely.

Reply:
```suggestion
* By configuration, tags can be made immutable, to prevent image tags from being overwritten.
```